core/comp_ui.py

```diff
@@ -149,8 +153,8 @@ class MinimalDisplay(_IDisplay):
   without testing it.
   """
   def __init__(self, comp_state, prompt_state, debug_f, num_lines_cap=10,
-               f=sys.stdout):
-    # type: (State, PromptState, _DebugFile, int, IO[bytes]) -> None
+               f=mylib.Stdout()):
```

Comment:
Hm default args don't get translated to anything; this is probably silently wrong

Also constructors are translated a bit differently than others

In a few places I had to use this `Init_*` method pattern, like `parse_ctx.Init_Trail()`

it is sort of a hack around keyword args.   You put the default initialization in the constructor, and then change the callers to call `Init_Field()` when they want to override

---

I'm  definitely open to better options!

Reply:
oh wow, nice catch! will do adopt `Init_*()` here

Reply:
Actually, it turns out nobody overrides these kwargs right now... so the latest commit just removes them and hard-codes the defaults. Figured we can add a way to set them in the future if/when we need it. What do you think?

Reply:
Yup sounds good!  This code probably has some generality that isn't used

----

BTW we have a hack for ONE optional arg on FUNCTIONS/methods if you ever need it, but I don't think it works for constructors

It's actually a hard-coded list in `mycpp/cppgen_pass.py` :-P    (another thing I want to get rid of, but it works)